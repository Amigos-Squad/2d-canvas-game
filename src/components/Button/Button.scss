@import 'scss/var';

.button {
  width: 100%;
  height: 40px;
  border-radius: 4px;
  border: none;
  outline: none;
  font-weight: 800;
  font-size: 20px;
  padding-top: 5px;
  cursor: pointer;
  background: $bg-light-color;
  color: $font-color;
  transition: background 0.2s linear, color 0.2s linear;
  text-transform: uppercase;

  &_primary {
    background: $accent-color;

    &:hover {
      background: $accent-dark-color;
    }

    &:focus {
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.7) inset;
    }
  }

  &_transparent {
    background: transparent;
    text-transform: uppercase;

    &:hover {
      color: $accent-color;
    }

    &:focus {
      color: $accent-dark-color;
    }
  }
}
