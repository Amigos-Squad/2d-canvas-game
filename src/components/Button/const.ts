export const enum BUTTON_TYPES {
  PRIMARY = 'primary',
  TRANSPARENT = 'transparent',
}
