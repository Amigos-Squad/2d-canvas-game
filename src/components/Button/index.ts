export { Button, BUTTON_TYPES } from './Button';
