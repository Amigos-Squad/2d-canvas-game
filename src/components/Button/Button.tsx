import React, { ReactElement, memo } from 'react';
import block from 'bem-cn-lite';
import { Props } from './Button.types';
import './Button.scss';

export enum BUTTON_TYPES {
  PRIMARY = 'primary',
  TRANSPARENT = 'transparent',
}

const buttonClass = block('button');

export const Button = memo(
  ({
    children,
    type,
    onClick,
    buttonType = BUTTON_TYPES.PRIMARY,
  }: Props): ReactElement => {
    const className = buttonClass({ [buttonType]: true });

    const clickHandler = (event: React.MouseEvent<HTMLButtonElement>) => {
      event.preventDefault();

      if (onClick) {
        onClick(event);
      }
    };

    return (
      <button type={type} className={className} onClick={clickHandler}>
        {children}
      </button>
    );
  }
);
