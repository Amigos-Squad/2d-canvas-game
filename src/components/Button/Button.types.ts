import { BUTTON_TYPES } from './const';

export type Props = {
  buttonType?: BUTTON_TYPES;
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  text?: string;
} & React.ButtonHTMLAttributes<HTMLButtonElement>;
