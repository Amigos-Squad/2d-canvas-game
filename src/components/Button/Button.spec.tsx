import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Button } from './Button';

describe('Компонент кнопки', () => {
  it('снапшот submit кнопки', () => {
    const tree = TestRenderer.create(
      <Button type="submit">LOGIN</Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('снапшот обычной кнопки', () => {
    const tree = TestRenderer.create(<Button>LOGIN</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
