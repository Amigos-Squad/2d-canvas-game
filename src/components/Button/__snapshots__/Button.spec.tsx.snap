// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент кнопки снапшот submit кнопки 1`] = `
<button
  className="button button_primary"
  onClick={[Function]}
  type="submit"
>
  LOGIN
</button>
`;

exports[`Компонент кнопки снапшот обычной кнопки 1`] = `
<button
  className="button button_primary"
  onClick={[Function]}
>
  LOGIN
</button>
`;
